### PR TITLE
IBC integration design doc

### PR DESCRIPTION
**Pull request by tzemanovic**
moved from https://github.com/anoma/anoma/pull/283

based on #266 

closes https://github.com/anoma/namada/issues/10

### CODE REVIEW

File: documentation/specs/src/interoperability/ibc.md [hunk 1]
```diff
@@ -1,32 +1,68 @@
...
-/balance/MINT_ADDR`. `{escrow_addr}`, `{BURN_ADDR}`, and `{MINT_ADDR}` are addresses of [`InternalAddress`](https://github.com/anoma/namada/blob/50b5e77f04a9afc036656353335bd232fcdba8a7/shared/src/types/address.rs). When these addresses are included in the changed keys after transaction execution, IBC token validity predicate is executed.
+A transaction escrowing/unescrowing a token changes the escrow account's balance of the token. The key is `{token_addr}/ibc/{port_id}/{channel_id}/balance/IbcEscrow`. A transaction burning a token changes the burn account's balance of the token. The key is `{token_addr}/ibc/{port_id}/{channel_id}/balance/IbcBurn`. A transaction minting a token changes the mint account's balance of the token. The key is `{token_addr}/ibc/{port_id}/{channel_id}/balance/IbcMint`. `IbcEscrow`, `IbcBurn`, and `IbcMint` are addresses of [`InternalAddress`](https://github.com/anoma/namada/blob/e3c2bd0b463b35d66fcc6d2643fd0e6509e03d99/core/src/types/address.rs#L446) and actually they are encoded in the storage key. When these addresses are included in the changed keys after transaction execution, IBC token validity predicate is triggered.
```
Comment (cwgoes) [inline documentation/specs/src/interoperability/ibc.md:23]: when we burn tokens, shouldn't we be reducing the supply? why do we send them to another account?

**Review by cwgoes — CHANGES_REQUESTED** [review]
One question

File: documentation/specs/src/interoperability/ibc.md (hunk 1 above)
Comment (yito88) [reply documentation/specs/src/interoperability/ibc.md:23]: The value of a key ending with `IbcBurn` or `IbcMint` isn't committed when a block commit. So, the burn reduces the supply. The keys have a value temporarily for validity predicates.

Comment (yito88) [reply documentation/specs/src/interoperability/ibc.md]: @cwgoes I added to explain it. Could you check it?

Comment (cwgoes) [reply documentation/specs/src/interoperability/ibc.md]: ah, gotcha

**Review by cwgoes — APPROVED** [review]

**Pull request merged**